homeassistant/components/lgthinq/coordinator.py

```diff
@@ -0,0 +1,164 @@
+"""DataUpdateCoordinator for the LG ThinQ device."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from thinqconnect import ConnectBaseDevice, DeviceType, ThinQApi, ThinQAPIException
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import COMPANY, DEVICE_TYPE_API_MAP, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DeviceDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """LG Device's Data Update Coordinator."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        device_api: ConnectBaseDevice,
+        *,
+        sub_id: str | None = None,
+    ) -> None:
+        """Initialize data coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{DOMAIN}_{device_api.device_id}",
+        )
+
+        # For washTower's washer or dryer
+        self.sub_id = sub_id
+
+        # The device name is usually set to 'alias'.
+        # But, if the sub_id exists, it will be set to 'alias {sub_id}'.
+        # e.g. alias='MyWashTower', sub_id='dryer' then 'MyWashTower dryer'.
+        self.device_name = (
+            f"{device_api.alias} {self.sub_id}" if self.sub_id else device_api.alias
+        )
+
+        # The unique id is usually set to 'device_id'.
+        # But, if the sub_id exists, it will be set to 'device_id_{sub_id}'.
+        # e.g. device_id='TQSXXXX', sub_id='dryer' then 'TQSXXXX_dryer'.
+        self.unique_id = (
+            f"{device_api.device_id}_{self.sub_id}"
+            if self.sub_id
+            else device_api.device_id
+        )
+
+        # Get the api instance.
+        self.device_api = device_api.get_sub_device(self.sub_id) or device_api
+
+    @property
+    def device_info(self) -> dr.DeviceInfo:
+        """Return the device information."""
+        return dr.DeviceInfo(
+            identifiers={(DOMAIN, self.unique_id)},
+            manufacturer=COMPANY,
+            model=self.device_api.model_name,
+            name=self.device_name,
+        )
```

Comment:
I think we can move this to `entity.py`, since we only use this there. You can use `_attr_device_info` to use in the constructor

Reply:
Modified.